nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialPropertyDescriptors.java

```diff
@@ -79,12 +80,65 @@ private CredentialPropertyDescriptors() { }
             .sensitive(true)
             .build();
 
+    private static final String DEFAULT_WORKLOAD_IDENTITY_SCOPE = "https://www.googleapis.com/auth/cloud-platform";
+    private static final String DEFAULT_WORKLOAD_IDENTITY_TOKEN_ENDPOINT = "https://sts.googleapis.com/v1/token";
+    private static final String DEFAULT_WORKLOAD_IDENTITY_SUBJECT_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:jwt";
+
+    public static final PropertyDescriptor WORKLOAD_IDENTITY_AUDIENCE = new PropertyDescriptor.Builder()
+            .name("Audience")
+            .description("The audience corresponding to the target Workload Identity Provider, typically the full resource name.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, AuthenticationStrategy.WORKLOAD_IDENTITY_FEDERATION.getValue())
+            .build();
+
+    public static final PropertyDescriptor WORKLOAD_IDENTITY_SCOPE = new PropertyDescriptor.Builder()
+            .name("Scope")
+            .description("OAuth2 scopes requested for the exchanged access token. Multiple scopes can be separated by space or comma.")
+            .required(true)
+            .defaultValue(DEFAULT_WORKLOAD_IDENTITY_SCOPE)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, AuthenticationStrategy.WORKLOAD_IDENTITY_FEDERATION.getValue())
+            .build();
+
+    public static final PropertyDescriptor WORKLOAD_IDENTITY_TOKEN_ENDPOINT = new PropertyDescriptor.Builder()
+            .name("STS Token Endpoint")
+            .description("Google Security Token Service endpoint used for token exchange.")
+            .required(false)
+            .defaultValue(DEFAULT_WORKLOAD_IDENTITY_TOKEN_ENDPOINT)
```

Comment:
`STS Token Endpoint` is effectively mandatory (it has a default value and is used in the code without null-check), so I would declare it with `required(true)` for the UI as well.